src/Identity/IdentityServer/ClientStore.cs

```diff
@@ -90,7 +90,12 @@ public async Task<Client> FindClientByIdAsync(string clientId)
 
     private async Task<Client> CreateApiKeyClientAsync(string clientId)
     {
-        var apiKey = await _apiKeyRepository.GetDetailsByIdAsync(new Guid(clientId));
+        if (!Guid.TryParse(clientId, out var guid))
+        {
+            return null;
```

Comment:
Question: What happens / what does the user see when `null` is returned?

Reply:
My thinking here was to replicate the behavior in the other client type parsing.  See https://github.com/bitwarden/server/blob/0f487f69005b092554d4b178645a03d4ef9a3478/src/Identity/IdentityServer/ClientStore.cs#L229 and https://github.com/bitwarden/server/blob/0f487f69005b092554d4b178645a03d4ef9a3478/src/Identity/IdentityServer/ClientStore.cs#L259, for example.

Also, checking the source code for IdentityServer4 you can see in [this](https://github.com/IdentityServer/IdentityServer4/blob/4dc10e665f5ede63274427036c25cdc216130eb9/src/EntityFramework.Storage/test/IntegrationTests/Stores/ClientStoreTests.cs#L34) test that `null` is the expected response when the client does not exist - as it would be in this case.